### PR TITLE
Added some UTs, got some UTs to work, minor fixes in code

### PR DESCRIPTION
**Pull request by gmuditdce**
UTs added and some minor code fixes for #196

### CODE REVIEW

File: cdr_plugin_folder_to_folder/processing/File_Processing.py [hunk 1]
```diff
@@ -444,7 +441,7 @@ def processDirectory (self, endpoint, dir, use_rebuild_zip=False):
                 self.finalize_not_supported(dir,hash)
             else:
                 self.finalize_failed(dir,hash)
-
+            return False
```
Comment (ggrig) [inline cdr_plugin_folder_to_folder/processing/File_Processing.py:444]: Remove this - not needed here.
The function returns False only for not expected cases

File: cdr_plugin_folder_to_folder/storage/Storage.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def hd2_status(self, path=''):
         return path_combine(self.config.hd2_status_location, path )  # add path and convert to absolute paths
 
     def hd3(self, path=''):
-        return path_combine(self.config.hd3_location, path        )  # add path and convert to absolute paths
+        return path_combine(self.config.hd3_location, os.path.basename(path))  # add path and convert to absolute paths
```
Comment (ggrig) [inline cdr_plugin_folder_to_folder/storage/Storage.py:100]: Remove this change - the function is used to obtain absolute paths to both files and folders in hd3

**Review by ggrig — CHANGES_REQUESTED** [review]

File: cdr_plugin_folder_to_folder/storage/Storage.py (hunk 2 above)
Comment (DinisCruz) [reply cdr_plugin_folder_to_folder/storage/Storage.py:100]: correct

Comment (gmuditdce) [reply cdr_plugin_folder_to_folder/storage/Storage.py:100]: @DinisCruz @ggrig 

Are there any cases apart from tests that will hit the else? I think no. If it does, and we need to use the path_combine function, it has a bug that needs to be resolved.

if path.startswith(self.config.hd1_location):
    rebuild_file_path = path.replace(self.config.hd1_location, self.config.hd3_location)
else:
    rebuild_file_path = self.storage.hd3(path)

One thing
def hd3(self, path=''):
    return path_combine(self.config.hd3_location, path)  # add path and convert to absolute paths

This is returning the same path as the input file location and rebuild is failing as src and destination files are the same file during copy of unsupported file to hd3 when save_unsupported_file_types is enabled

For the code
def hd3(self, path=''):
    return path_combine(self.config.hd3_location, path)  # add path and convert to absolute paths

When
self.config.hd3_location = /tmp/tmphqw5nar2/hd3
path = /tmp/tmpcavfwsyn.tmp

It returns
/tmp/tmpcavfwsyn.tmp

In the integration tests dir there is a test that reproduces this behaviour
def test_processBadFile(self):
    bad_file = temp_file(contents=random_text())
    metadata = self.meta_service.create_metadata(bad_file)
    endpoint = f'http://{self.sdk_server}:{self.sdk_port}'
    dir = metadata.metadata_folder_path()
    try:
        result = self.file_processing.processDirectory(endpoint=endpoint, dir=dir)
    except Exception as e:
        traceback.print_exc()
        self.fail("Should not have thrown")
    assert result == True
    metadata.load()
    assert metadata.data.get('rebuild_status') == 'The file type is not currently supported'

**Review by ggrig — APPROVED** [review]

**Pull request merged**